lib/pbench/server/indexer.py

```diff
@@ -1536,121 +1539,113 @@ def _make_source_json(self):
         We only process the top-level result.json file since it has all the
         data we need in one place.
         """
-        for dirname in self.json_dirs:
-            if dirname.startswith("sample"):
-                # sample result.json - we ignore each sample result.json since
-                # each sample is contained in the experiment level
-                # result.json.
-                continue
-            elif dirname != self.ptb.dirname:
-                # iteration result.json - there is no value in processing the
-                # result.json file at the iteration level because everything
-                # it contains is also in the experiment level's result.json.
-                continue
+        result_json = os.path.join(
+            self.ptb.extracted_root, self.ptb.dirname, "result.json"
+        )
+        try:
+            # Read the file and interpret it as a JSON document.
+            with open(result_json) as fp:
+                results = json.load(fp)
+        except Exception as e:
+            self.logger.warning(
+                "result-data-indexing: encountered invalid JSON file," " {}: {:r} ({})",
```

Comment:
Or this might be where `black` is unhappy. First off, if you're joining the continuation string onto the same line, you don't need to terminate the first string, which would save you 3 characters. Not sure that's enough, though.

Reply:
yeah weird I am not sure why black is throwing error in the unittest, when in fact it says nothing to reformat when I run the indexer.py file with black